sonic_platform_base/sonic_sfp/sfputilhelper.py

```diff
@@ -83,24 +89,25 @@ def read_porttab_mappings(self, porttabfile, asic_inst=0):
             # Mapping of logical port names available on a system to ASIC instance
             self.logical_to_asic[intf_name] = asic_inst
 
-        self.logical = logical
-        self.logical_to_physical = logical_to_physical
-        self.physical_to_logical = physical_to_logical
+        self.logical.extend(logical)
+        self.logical = list(set(self.logical))
+        self.logical_to_physical.update(logical_to_physical)
+        self.physical_to_logical.update(physical_to_logical)
 
         return None
 
 
     def read_all_porttab_mappings(self, platform_dir, num_asic_inst):
```

Comment:
@mlok-nokia can you test this in non-multi-asic platform?

Reply:
We don't have the non-multi-asic platform to test this change.  But there is an existing  UT test_read_port_mappings() for the it.